### ISSUE TITLE
Consider a section clarifying that component assets are internal

### ISSUE THREAD

**Issue by jcfranco** [Esri/calcite-design-system#10439]
### Description

Assets, such as icon path data and translations, should be treated as internal and not qualify for breaking change consideration. 

For example, issues like [this](https://github.com/Esri/calcite-design-system/issues/10438) should be able to be simple refactors between minor, patch releases.

### Notes

* We do [document our assets path](https://developers.arcgis.com/calcite-design-system/get-started/#load-the-assets),
* `messageOverrides` would probably need to hold on to entries that correspond to removed messages until a breaking change release as these might be breaking if used. There might be a way to widen the type to avoid breakage.

### Which Component

Translated components + `calcite-icon`.

**Comment by geospatialem** [comment]
cc @DitwanP @macandcheese added the `Needs milestone` label to this spike effort with the Maps SDK for JS improved guidance for 25.R2 update. ✨🤜🏻🤛🏻✨

**Comment by isaacbraun** [comment]
Apologies that this has taken a while, and if I'm misunderstanding this.

@Esri/calcite-pes, @jcfranco, could I get some more context on what actions are needed for this spike? Is it researching if/how we can document that the assets are internal via JSDoc, as well as the widening of types, or is it something else entirely? Or is this not a spike but rather a documentation effort? TIA 🤜💥🤛

**Comment by geospatialem** [comment]
> Apologies that this has taken a while, and if I'm misunderstanding this.
> 
> [@Esri/calcite-pes](https://github.com/orgs/Esri/teams/calcite-pes), [@jcfranco](https://github.com/jcfranco), could I get some more context on what actions are needed for this spike? Is it researching if/how we can document that the assets are internal via JSDoc, as well as the widening of types, or is it something else entirely? Or is this not a spike but rather a documentation effort? TIA 🤜💥🤛

@isaacbraun Yes, great question! _Believe_ (its been a year-ish 😆) this relates to the `messageOverrides` that we are working on improving translation updates as they come in from the Translations team **and** for any `icon` changes that take place (_this one may not be an issue since we've coordinated on breaking changes with the Icons teams_). 

Since we exclude the `messageOverrides` strings from breaking changes and icons could be modified outside of breaking changes (_perhaps no longer an issue based on team discussions_) the above would be to consider approaches of documenting/communicating this pattern, where some of the following _could_ be considered (among others):
- We could make changes to string values or icons outside of our breaking change releases
- _Optionally_ inform our users that these changes could take place at any time
- _Optionally_ provide additional guidance for working with `messageOverrides` / `icon` and expectations for support

**Comment by isaacbraun** [comment]
Thank you for the breakdown, @geospatialem! Apologies for my delay in response 😅

I was able to dig up the PR where the Maps SDK for JS team updated their handling of this situation ([#5510](https://devtopia.esri.com/WebGIS/arcgis-web-components/pull/5510)) along with some other findings, and think we have a pretty easy path forward.

 - The Maps SDK for JS team decided that:
 	- The doc should clearly allow string change issues in between releases, stating that users should use at their own risk.
 	- The doc could point to the .en.json in the CDN, pointing users to figure out what strings to use for what there. **This has not been done** as far as I can tell.
 - Our typing already uses `Record<string, unknown> | undefined`, which should not prevent us from making changes during minor/patch releases. 

The SDK team implemented their warning via `copyDoc` as below (example with [arcgis-bookmarks](https://next.gha.afd.arcgis.com/javascript/latest/references/map-components/components/arcgis-bookmarks/#messageOverrides)):

<img width="934" height="184" alt="Image" src="https://github.com/user-attachments/assets/59ef1c20-1cf3-4ced-80f2-6815264801d3" />

Adopting this messaging would allow us to change strings when needed, outside of breaking change releases, and as far as I can tell there is no issues with the current typing.

I have few questions before proceeding:
1. Does this direction cover our concerns?
2. Should we link/point to the `.en.json` CDN file in the doc string? I lean towards no as the Maps SDK team has not, and we can wait for user feedback.
3. Do we have any remaining concerns with Icons?

cc @jcfranco

**Comment by geospatialem** [comment]
Awesome [summary above](https://github.com/Esri/calcite-design-system/issues/10439#issuecomment-4836531896), @isaacbraun! 💪🏻  Really like the direction and coordination with the SDK with a streamlined comment without taking away too much focus on the intent of the `messageOverrides` property and its description. 🏆 

> 1. Does this direction cover our concerns?

_Believe_ this to be the case, and love this small snippet of context for folks, it definitely communicates messaging without taking away from content.

>2. Should we link/point to the .en.json CDN file in the doc string? I lean towards no as the Maps SDK team has not, and we can wait for user feedback.

Concur with your thoughts on not linking, this could be a good topic to coordinate with for a larger meeting with the SDK and i18n/l18n team later this year.


> 3. Do we have any remaining concerns with Icons?

Would defer to @jcfranco on the icons specifically, either way we should likely communicate our direction with the Icons team internally so they are aware of the direction moving forward.

**Comment by jcfranco** [comment]
Great breakdown. Here's my take:

1. The message changes not being considered breaking between releases is covered, but the assets aspect is still pending. Specifically, we should reserve the right to change the structure of the default assets without considering it a breaking change (see ⁠[assets directory](https://app.unpkg.com/@esri/calcite-components@5.1.2/files/dist/cdn/assets) and ⁠[icon assets directory](https://app.unpkg.com/@esri/calcite-components@5.1.2/files/dist/cdn/assets/icon)).
2. Agreed. The component playground provides the keys and values already, so users can effectively override messages.
3. I don't think this impacts the Icons team. This concern is more about the asset path perspective, which is exclusive to components.

**Comment by isaacbraun** [comment]
Thanks for the feedback! With some additional chatting offline, this spike is complete:
- Updating this issue to align our `messageOverrides` property description with the SDK.
- Creating an issue in our documentation repo to add a note that we "reserve the right to change the structure of the default assets without considering it a breaking change."

**Comment by DitwanP** [comment]
Verified 🍠✍️✨
